resources/docdb-cluster.go

```diff
@@ -61,15 +77,20 @@ func (l *DocDBClusterLister) List(ctx context.Context, o interface{}) ([]resourc
 }
 
 type DocDBCluster struct {
-	svc *docdb.Client
+	svc      *docdb.Client
+	settings *libsettings.Setting
 
 	ID                 *string
 	DeletionProtection *bool
 	Tags               []docdbtypes.Tag
 }
 
+func (r *DocDBCluster) Settings(settings *libsettings.Setting) {
+	r.settings = settings
+}
+
 func (r *DocDBCluster) Remove(ctx context.Context) error {
-	if *r.DeletionProtection {
+	if r.settings.GetBool("DisableDeletionProtection") {
```

Comment:
If you check that `DeletionProtection` is true, then we can avoid an API call that's not necessary